src/access/wac.ts

```diff
@@ -314,10 +337,26 @@ async function setActorAccess<T extends WithServerResourceInfo>(
     resourceWithAcl,
     resourceAcl
   );
-  const currentAccess = readAccess(resourceWithOldAcl, actor) as AclAccess;
-  const wacAccess = universalAccessToAcl(access, currentAccess);
 
-  const updatedResourceAcl = writeAccess(resourceAcl, actor, wacAccess);
+  let currentAccess: AclAccess;
+  let updatedResourceAcl: (AclDataset & WithChangeLog) | undefined = undefined;
+  let wacAccess: AclAccess | undefined = undefined;
+
+  if (actor !== undefined) {
```

Comment:
Since there are three different places this function is called form, and we're special casing to distinguish between one and the two others, it doesn't feel that generic anymore, so... Maybe this isn't the right abstraction?

I wonder if maybe it would be better to just have the first part of the function be a standalone function, and to have everything from this `if` statement on just happen in the calling functions. (Or potentially, the final `try..catch` block in a separate function as well.) What do you think?

(Also, I just realised that `readAccess` can be a bit confusing given that "read" is also an access mode :) If we do keep it, maybe it should be `getAccess` or something like that?)

Reply:
c685ad7893dec1789efbe2d4b5f8e2858ae723d2